dask/array/overlap.py

```diff
@@ -23,6 +23,80 @@
 from .creation import empty_like, full_like
 
 
+class ArrayOverlapLayer(Layer):
+    """Simple HighLevelGraph array overlap layer.
+
+    Lazily computed High-level graph layer for a array overlap operations.
+
+    Parameters
+    ----------
+    name : str
+        Name of new output overlap array.
+    """
+
+    def __init__(
+        self,
+        name,
+    ):
+        self.name = name
+
+    def __repr__(self):
+        return "ArrayOverlapLayer<name='{}'".format(self.name)
+
+    @property
+    def _dict(self):
+        """Materialize full dict representation"""
+        if hasattr(self, "_cached_dict"):
+            return self._cached_dict
+        else:
+            dsk = self._construct_graph()
+            self._cached_dict = dsk
+        return self._cached_dict
+
+    def __getitem__(self, key):
+        return self._dict[key]
+
+    def __iter__(self):
+        return iter(self._dict)
+
+    def __len__(self):
+        return len(self._dict)
```

Comment:
Could we plan on these not needing to materialize `self._dict`? Maybe we'd want a [`get_output_keys()`](https://github.com/dask/dask/blob/911dc6641144e1fcff2cc0926214ee81d49d2a28/dask/blockwise.py#L312-L323) method similar to Blockwise. In particular, `__len__` gets called by `_repr_html_`, so materializing the graph in there can make display in a notebook slow.

Reply:
Those are good things to flag, thanks @gjoseph92